x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -22,7 +22,11 @@ import { registerDatasource } from './applications/ingest_manager/sections/agent
 
 export { IngestManagerConfigType } from '../common/types';
 
-export type IngestManagerSetup = void;
+// We need to provide an object instead of void so that dependent plugins know when Ingest Manager
+// is disabled.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IngestManagerSetup {}
```

Comment:
I didn't realize the current code would prevent consumers from even testing if it was enabled

Reply:
Thanks for taking a look and approving @jfsiii! I doubt anyone was aware of this type of interaction. I stumbled on it by accident myself! I opened https://github.com/elastic/kibana/issues/69845 to discuss the problem and seek a global solution.